packages/shylu/shylu_dd/frosch/src/Tools/FROSch_Tools_def.hpp

```diff
@@ -1035,29 +1174,33 @@ namespace FROSch {
          */
         unsigned numVec = multiVector->getNumVectors();
         Array<unsigned> arrayZero(0);
+        RCP<const Map<LO,GO,NO> > multiVectorMap = multiVector->getMap();
         RCP<MultiVector<SC,LO,GO,NO> > resultMultiVector;
         if (numVec>0) {
             unsigned itmp = 0;
             SC en = ScalarTraits<SC>::zero();
             SC de = ScalarTraits<SC>::zero();
             SC norm = ScalarTraits<SC>::zero();
-            RCP<MultiVector<SC,LO,GO,NO> > tmpMultiVector = MultiVectorFactory<SC,LO,GO,NO>::Build(multiVector->getMap(),numVec);
+            RCP<MultiVector<SC,LO,GO,NO> > tmpMultiVector = MultiVectorFactory<SC,LO,GO,NO>::Build(multiVectorMap,numVec);
             for (unsigned i=0; i<numVec; i++) {
-                tmpMultiVector->getVectorNonConst(i-itmp)->update(ScalarTraits<SC>::one(),*multiVector->getVector(i),ScalarTraits<SC>::zero());
+                RCP<const Vector<SC,LO,GO,NO> > multiVector_i = multiVector->getVector(i);
+                RCP<Vector<SC,LO,GO,NO> > tmpMultiVectorNonConst_i_itmp = tmpMultiVector->getVectorNonConst(i-itmp);
+                tmpMultiVectorNonConst_i_itmp->update(ScalarTraits<SC>::one(),*multiVector_i,ScalarTraits<SC>::zero());
                 for (unsigned j=0; j<i-itmp; j++) {
-                    en = tmpMultiVector->getVector(i-itmp)->dot(*tmpMultiVector->getVector(j));
-                    de = tmpMultiVector->getVector(j)->dot(*tmpMultiVector->getVector(j));
-                    tmpMultiVector->getVectorNonConst(i-itmp)->update(-en/de,*tmpMultiVector->getVector(j),ScalarTraits<SC>::one());
+                    RCP<const Vector<SC,LO,GO,NO> > tmpMultiVector_j = tmpMultiVector->getVector(j);
+                    en = tmpMultiVectorNonConst_i_itmp->dot(*tmpMultiVector_j);
+                    de = tmpMultiVector_j->dot(*tmpMultiVector_j);
+                    tmpMultiVectorNonConst_i_itmp->update(-en/de,*tmpMultiVector_j,ScalarTraits<SC>::one());
                 }
-                norm = tmpMultiVector->getVector(i-itmp)->norm2();
+                norm = tmpMultiVectorNonConst_i_itmp->norm2();
                 if (norm<1.0e-10) {
                     arrayZero.push_back(i);
                     itmp++;
                 } else {
                     //tmpMultiVector->getVectorNonConst(i-itmp)->scale(1.0/norm);
```

Comment:
I'm not sure what's going on here, but if you really don't need to scale the multivector, then you should just delete the else branch and comment, to reassure readers that you didn't forget anything here :-) .